File: app/local_plugins/vendors/mcp/ios/Sources/mcp/NativeMcpService.swift
import Foundation

/// MCP（Model Context Protocol）服务原生接口（iOS 端）
///
/// 与 Kotlin `NativeMcpService` 对齐。单实例对应单 server。
///
/// 本文件目前**未启用**：iOS native 链路（agent_chat/ios、llm_openai/ios、
/// ai_plugin_interface/ios）尚未铺开，因此该协议没有对应的 podspec/registry
/// 集成。等 iOS native 链路整体启动时把它接进 ai_plugin_interface 的 iOS 模块。
///
/// 生命周期：`uninitialized → initialize(_:) → ready → callTool/listTools → dispose → disposed`
public protocol NativeMcpService: AnyObject {

    /// 握手并就绪
    /// - Parameter configJson: 序列化后的 McpServerConfig
    /// - Throws: 握手失败时抛出，调用方据此把该 server 标记为不可用
    func initialize(configJson: String) async throws

    /// 拉取该 server 的工具列表（按 enabledTools 白名单过滤）
    /// - Returns: `[{"name","description","inputSchema":[String:Any?],"serverId"}]`
    func listTools() async throws -> [[String: Any?]]

    /// 调用工具
    /// - Parameters:
    ///   - toolName: 工具名
    ///   - argsJson: JSON 字符串形式的参数对象
    /// - Returns: `{"content":String, "isError":Bool}`
    func callTool(toolName: String, argsJson: String) async throws -> [String: Any?]

    /// 释放资源
    func dispose()
}

File: apps/client/local_plugins/core/ai_plugin_interface/android/src/main/kotlin/com/aiagent/plugin_interface/NativeMcpService.kt
package com.aiagent.plugin_interface

/**
 * MCP（Model Context Protocol）服务原生接口
 *
 * 实现方：mcp_streamable_http 等传输厂商插件
 * 使用方：agent_chat 等 Agent 类型插件
 *
 * 单实例对应单 server。多 server 聚合由 agent 容器内的 McpRouter 负责。
 *
 * 生命周期：`uninitialized → initialize(...) → ready → callTool/listTools → dispose → disposed`
 */
interface NativeMcpService {

    /**
     * 握手并就绪
     *
     * @param configJson  序列化后的 McpServerConfig（id/name/url/authHeader/enabledTools/timeoutSeconds/extraHeaders）
     * @throws Exception  握手失败时抛出，调用方据此把该 server 标记为不可用
     */
    suspend fun initialize(configJson: String)

    /**
     * 拉取该 server 的工具列表（按 enabledTools 白名单过滤）
     *
     * @return List<Map> 每条 = {"name", "description", "inputSchema":Map, "serverId"}
     */
    suspend fun listTools(): List<Map<String, Any?>>

    /**
     * 调用工具
     *
     * @param toolName  工具名
     * @param argsJson  JSON 字符串形式的参数对象
     * @return Map  {"content": String, "isError": Boolean}
     */
    suspend fun callTool(toolName: String, argsJson: String): Map<String, Any?>

    /** 释放资源（关闭 OkHttp client、清理 sessionId 等） */
    fun dispose()
}
